python/yugabyte_db_thirdparty/library_checking.py

```diff
@@ -110,18 +189,34 @@ def run(self) -> None:
                 for candidate in candidate_dirs:
                     if dir_pattern.match(candidate.name):
                         examine_path = os.path.join(installed_dir, candidate.name)
-                        for dirpath, dirnames, files in os.walk(examine_path):
+                        for dirpath, dir_names, files in os.walk(examine_path):
                             for file_name in files:
                                 full_path = os.path.join(dirpath, file_name)
                                 if os.path.islink(full_path):
                                     continue
-                                if not self.check_libs_for_file(full_path):
-                                    test_pass = False
+                                if not self.should_check_file(full_path):
+                                    continue
+
+                                self.files_to_check.append(full_path)
+
+        self.before_checking_all_files()
+        test_pass = self.check_all_files()
+
         if not test_pass:
             fatal(f"Found problematic library dependencies, using tool: {self.tool}")
         else:
             log("No problems found with library dependencies.")
 
+    def before_checking_all_files(self) -> None:
+        pass
+
+    def check_all_files(self) -> bool:
+        success = True
```

Comment:
(matter of taste, but short circuits automatically) `return all(map(self.check_libs_for_file, self.files_to_check))`

Reply:
See my earlier comment. I don't want to return early from this loop.